<?xml version="1.0"?>
<robot xmlns:xacro="http://www.ros.org/wiki/xacro">

  <xacro:macro name="titati_ros2control" params="prefix">
    <ros2_control name="WebotsControl" type="system">
      <hardware>
        <plugin>titati_webots_ros2_control::WebotsBridge</plugin>
      </hardware>
      <joint name="${prefix}joint_frontleft_leg_1">
        <!-- <command_interface name="position"/> -->
        <!-- <command_interface name="velocity"/> -->
        <command_interface name="effort"/>
        <state_interface name="position"/>
        <state_interface name="velocity"/>
        <state_interface name="effort"/>
        <!-- <param name="initial_position">0.0</param> -->
        <!-- initial position for the FakeSystem -->
      </joint>
      <joint name="${prefix}joint_frontleft_leg_2">
        <!-- <command_interface name="position"/>
            <command_interface name="velocity"/> -->
        <command_interface name="effort"/>
        <state_interface name="position"/>
        <state_interface name="velocity"/>
        <state_interface name="effort"/>
        <param name="initial_position">0.0</param>
        <!-- initial position for the FakeSystem -->
      </joint>
      <joint name="${prefix}joint_frontleft_leg_3">
        <!-- <command_interface name="position"/>
            <command_interface name="velocity"/> -->
        <command_interface name="effort"/>
        <state_interface name="position"/>
        <state_interface name="velocity"/>
        <state_interface name="effort"/>
        <param name="initial_position">0.0</param>
        <!-- initial position for the FakeSystem -->
      </joint>
      <joint name="${prefix}joint_frontleft_leg_4">
        <!-- <command_interface name="position"/>
            <command_interface name="velocity"/> -->
        <command_interface name="effort"/>
        <state_interface name="position"/>
        <state_interface name="velocity"/>
        <state_interface name="effort"/>
        <param name="initial_position">0.0</param>
        <!-- initial position for the FakeSystem -->
      </joint>
      <joint name="${prefix}joint_frontright_leg_1">
        <!-- <command_interface name="position"/>
            <command_interface name="velocity"/> -->
        <command_interface name="effort"/>
        <state_interface name="position"/>
        <state_interface name="velocity"/>
        <state_interface name="effort"/>
        <!-- <param name="initial_position">0.0</param> -->
        <!-- initial position for the FakeSystem -->
      </joint>
      <joint name="${prefix}joint_frontright_leg_2">
        <!-- <command_interface name="position"/>
            <command_interface name="velocity"/> -->
        <command_interface name="effort"/>
        <state_interface name="position"/>
        <state_interface name="velocity"/>
        <state_interface name="effort"/>
        <param name="initial_position">0.0</param>
        <!-- initial position for the FakeSystem -->
      </joint>
      <joint name="${prefix}joint_frontright_leg_3">
        <!-- <command_interface name="position"/>
            <command_interface name="velocity"/> -->
        <command_interface name="effort"/>
        <state_interface name="position"/>
        <state_interface name="velocity"/>
        <state_interface name="effort"/>
        <param name="initial_position">0.0</param>
        <!-- initial position for the FakeSystem -->
      </joint>
      <joint name="${prefix}joint_frontright_leg_4">
        <!-- <command_interface name="position"/>
            <command_interface name="velocity"/> -->
        <command_interface name="effort"/>
        <state_interface name="position"/>
        <state_interface name="velocity"/>
        <state_interface name="effort"/>
        <param name="initial_position">0.0</param>
        <!-- initial position for the FakeSystem -->
      </joint>
      <joint name="${prefix}joint_rearleft_leg_1">
        <!-- <command_interface name="position"/> -->
        <!-- <command_interface name="velocity"/> -->
        <command_interface name="effort"/>
        <state_interface name="position"/>
        <state_interface name="velocity"/>
        <state_interface name="effort"/>
        <!-- <param name="initial_position">0.0</param> -->
        <!-- initial position for the FakeSystem -->
      </joint>
      <joint name="${prefix}joint_rearleft_leg_2">
        <!-- <command_interface name="position"/>
            <command_interface name="velocity"/> -->
        <command_interface name="effort"/>
        <state_interface name="position"/>
        <state_interface name="velocity"/>
        <state_interface name="effort"/>
        <param name="initial_position">0.0</param>
        <!-- initial position for the FakeSystem -->
      </joint>
      <joint name="${prefix}joint_rearleft_leg_3">
        <!-- <command_interface name="position"/>
            <command_interface name="velocity"/> -->
        <command_interface name="effort"/>
        <state_interface name="position"/>
        <state_interface name="velocity"/>
        <state_interface name="effort"/>
        <param name="initial_position">0.0</param>
        <!-- initial position for the FakeSystem -->
      </joint>
      <joint name="${prefix}joint_rearleft_leg_4">
        <!-- <command_interface name="position"/>
            <command_interface name="velocity"/> -->
        <command_interface name="effort"/>
        <state_interface name="position"/>
        <state_interface name="velocity"/>
        <state_interface name="effort"/>
        <param name="initial_position">0.0</param>
        <!-- initial position for the FakeSystem -->
      </joint>
      <joint name="${prefix}joint_rearright_leg_1">
        <!-- <command_interface name="position"/>
            <command_interface name="velocity"/> -->
        <command_interface name="effort"/>
        <state_interface name="position"/>
        <state_interface name="velocity"/>
        <state_interface name="effort"/>
        <!-- <param name="initial_position">0.0</param> -->
        <!-- initial position for the FakeSystem -->
      </joint>
      <joint name="${prefix}joint_rearright_leg_2">
        <!-- <command_interface name="position"/>
            <command_interface name="velocity"/> -->
        <command_interface name="effort"/>
        <state_interface name="position"/>
        <state_interface name="velocity"/>
        <state_interface name="effort"/>
        <param name="initial_position">0.0</param>
        <!-- initial position for the FakeSystem -->
      </joint>
      <joint name="${prefix}joint_rearright_leg_3">
        <!-- <command_interface name="position"/>
            <command_interface name="velocity"/> -->
        <command_interface name="effort"/>
        <state_interface name="position"/>
        <state_interface name="velocity"/>
        <state_interface name="effort"/>
        <param name="initial_position">0.0</param>
        <!-- initial position for the FakeSystem -->
      </joint>
      <joint name="${prefix}joint_rearright_leg_4">
        <!-- <command_interface name="position"/>
            <command_interface name="velocity"/> -->
        <command_interface name="effort"/>
        <state_interface name="position"/>
        <state_interface name="velocity"/>
        <state_interface name="effort"/>
        <param name="initial_position">0.0</param>
        <!-- initial position for the FakeSystem -->
      </joint>
      <sensor name="imu">
        <state_interface name="orientation.x"/>
        <state_interface name="orientation.y"/>
        <state_interface name="orientation.z"/>
        <state_interface name="orientation.w"/>
        <state_interface name="angular_velocity.x"/>
        <state_interface name="angular_velocity.y"/>
        <state_interface name="angular_velocity.z"/>
        <state_interface name="linear_acceleration.x"/>
        <state_interface name="linear_acceleration.y"/>
        <state_interface name="linear_acceleration.z"/>
      </sensor>
    </ros2_control>

  </xacro:macro>

</robot>